src/probnum/filtsmooth/gaussfiltsmooth/iterated_component.py

```diff
@@ -0,0 +1,159 @@
+    >>> print(out_iterated.mean)
+    [ 0.18201288  0.45367681 -9.1948478 ]
+    """
+
+    def __init__(
+        self,
+        component,
+        stopcrit=None,
+    ):
+        self._component = component
+        self.stopcrit = StoppingCriterion() if stopcrit is None else stopcrit
+        super().__init__(input_dim=component.input_dim, output_dim=component.output_dim)
+
+    # Iterated filtering implementation
+
+    def backward_rv(
+        self,
+        rv_obtained,
+        rv,
+        rv_forwarded=None,
+        gain=None,
+        t=None,
+        dt=None,
+        _diffusion=1.0,
+        _linearise_at=None,
+    ):
+        current_rv, info = self._component.backward_rv(
+            rv_obtained=rv_obtained,
+            rv=rv,
+            t=t,
+            dt=dt,
+            _diffusion=_diffusion,
+            _linearise_at=_linearise_at,
+        )
+
+        new_mean = current_rv.mean.copy()
+        old_mean = np.inf * np.ones(current_rv.mean.shape)
+        while not self.stopcrit.terminate(
+            error=new_mean - old_mean, reference=new_mean
+        ):
+            old_mean = new_mean.copy()
+            current_rv, info = self._component.backward_rv(
+                rv_obtained=rv_obtained,
+                rv=rv,
+                t=t,
+                dt=dt,
+                _diffusion=_diffusion,
+                _linearise_at=current_rv,
+            )
+            new_mean = current_rv.mean.copy()
+        return current_rv, info
+
+    def backward_realization(
+        self,
+        real_obtained,
+        rv,
+        rv_forwarded=None,
+        gain=None,
+        t=None,
+        dt=None,
+        _diffusion=1.0,
+        _linearise_at=None,
+    ):
+        return self._backward_realization_via_backward_rv(
+            real_obtained,
+            rv=rv,
+            rv_forwarded=rv_forwarded,
+            gain=gain,
+            t=t,
+            dt=dt,
+            _diffusion=_diffusion,
+            _linearise_at=_linearise_at,
+        )
+
+    # These need to be re-implemented here, because otherwise this class
+    # cannot be instantiated (abc things)
+
+    def forward_rv(
+        self, rv, t, dt=None, compute_gain=False, _diffusion=1.0, _linearise_at=None
+    ):
+        return self._component.forward_rv(
+            rv,
+            t,
+            dt=dt,
+            compute_gain=compute_gain,
+            _diffusion=_diffusion,
+            _linearise_at=_linearise_at,
+        )
+
+    def forward_realization(
+        self, real, t, dt=None, compute_gain=False, _diffusion=1.0, _linearise_at=None
+    ):
+        return self._component.forward_realization(
+            real,
+            t,
+            dt=dt,
+            compute_gain=compute_gain,
+            _diffusion=_diffusion,
+            _linearise_at=_linearise_at,
+        )
+
+    # Pass on all the rest to the EKF/UKF component
+
+    def __getattr__(self, attr):
```

Comment:
All of this feels a bit hacky :sweat_smile: But I couldn't figure out how to do things differently, I thought class decorators might fix this but idk.

Reply:
Which means I'm fine with keeping things this way :+1: And in the case that you also don't like this part we could think about alternatives together.

Reply:
Isnt this basically  class decorator? In any case, I just wanted something which (a) essentially inherits everyhing but backward_rv from the component that it passed to `IteratedComponent` (e.g. EKF **or** UKF), and (b) has as little code as possible. If you have any other idea, please let me know. We can also talk about this separately, or (hereby) ask @schmidtjonathan's  opinion ;)

Reply:
I think this should be implemented as a wrapper.
I have an idea how this could be implemented elegantly, but I would suggest opening a separate PR for this.